### PR TITLE
docs: Update link to USERS.md in README from RAW Github to standard Github UI

### PR DESCRIPTION
**Pull request by ondrejsika**
Fix link

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -308,7 +308,7 @@ How they are selected and govern is outlined in our `governance document <https:
 Adopters
 --------
 A list of adopters of the Cilium project who are deploying it in production, and of their use cases,
-can be found in file `USERS.md <https://raw.githubusercontent.com/cilium/cilium/main/USERS.md>`__.
+can be found in file `USERS.md <https://github.com/cilium/cilium/blob/main/USERS.md>`__.
```
Comment (sayboras) [inline README.rst:311]: The commit message was to fix link, but the existing link is working as expected, probably it's about using github.com link instead, then I am not sure about the value proposition for this change :thinking:

**Review by justankiit — COMMENTED** [review]
It will be great, if you can refer a issue for which PR has been created!
Also, when you have raise the PR there would be a release-note-label, then only this PR would go ahead!

File: README.rst (hunk 1 above)
Comment (ti-mo) [reply README.rst:311]: @sayboras The `raw.githubusercontent.com` link results in a raw markdown document displayed in the browser, while the new link proposed here renders the markdown displayed through the GH UI. Much nicer IMO.

**Review by ti-mo — APPROVED** [review]
My approval won't help this along, but this is a really straightforward fix. Let's get it merged!

File: README.rst (hunk 1 above)
Comment (sayboras) [reply README.rst:311]: ah right :heavy_check_mark:, it's personal preference I can say. I am good with either case, but the commit message could be clearer.

**Review by justankiit — APPROVED** [review]
now fixed!

**Review by sayboras — APPROVED** [review]
My approval won't help either, it's better to leave it to docs team to decide if we want to move ahead.

**Review by ghost — APPROVED** [review]
looks good

**Comment by ti-mo** [comment]
@ondrejsika No merge commits, please. Simply rebase on top of `main` and force-push.

**Comment by joestringer** [comment]
Filed https://github.com/cilium/cilium/issues/31147 for the ci-ginkgo failure.

**Pull request merged**